### DISCUSSION TITLE
Model in NoData State

### DISCUSSION THREAD

**Discussion by xMikeTR** [TabularEditor/TabularEditor3#1369] [🙋♀️TE3 Q&A] [answered]
Hey everyone,

Hoping you can help me with this,

I've connected tabular editor to a Databricks SQL warehouse and imported some tables.

When doing so it said the model is in NoData state and on refresh it gives me the following error:
Object	Description	Progress	Duration
Error	"{""error"":{""code"":""DMTS_MonikerWithUnboundDataSources"",""pbi.error"":{""code"":""DMTS_MonikerWithUnboundDataSources"",""details"":[{""code"":""Url"",""detail"":{""type"":1,""value"":""Databricks""}},{""code"":""ConnectionType"",""detail"":{""type"":0,""value"":""Extension""}}],""exceptionCulprit"":1}}}

Technical Details:
RootActivityId: 2bb80936-0052-4718-9314-efdbc94bfe3a
Date (UTC): 9/2/2024 8:22:24 AM"		

Working on just Metadata and deploying through pipeline, we run into an issue of memory when it's time to refresh.

So having the actual data and being able to work on it in TE, it would allow us to apply incremental refresh and create proper partitions.

Any help is appreciated,

Thanks

**Comment by mlonsk** [comment]
Hi @xMikeTR 
Could you give a few more details about your model.
- Are you deploying to SSAS/AAS or Power BI workspace
- What type of data source have you created? (Legacy, Power Query or Implicit)
- Does import table wizard and schema compare work inside Tabular Editor 3?
- How have you set up the credentials for the data source on the server (SSAS/AAS or Power BI)

Solely from your message it could look like the credentials are not setup correctly, but its hard to say for sure.

**Comment by xMikeTR** [comment]
Hey , thanks for the reply.

I am deploying to Power BI Workspace,

When I imported I chose the option of implicit data sources and then use the hostname and http path from our databricks cluster.
"Does import table wizard and schema compare work inside Tabular Editor 3?"
Sorry I don't follow, can you explain?

Credentials are managed by AD, and we simply authenticate.

I have tried a different approach and on import I selected to use Structured(Power Query) Source and through here, using Simba Spark and proper connection string. 
It's still on NoData state and on refresh , this time it says one or more data sources does not have credentials specified in Power BI Service.

Upon checking PowerBI service , I see it has no gateway. I'm currently trying to check with who setup the gateways but, 
You think this is the issue? or is there another solution I'm not seeing?

Appreciate your time

**Reply by mlonsk** [reply, marked as answer]
Hi @xMikeTR 
To check, are you using the latest version of Tabular Editor 3 (3.17.1)?
We added a new Databricks connector in Tabular Editor 3, depending on whether you connect to Azure or other providers. If you are on an earlier version and hosting Databricks in a place other than Azure, choosing the new connector could perhaps help you.  You can read about it in [our release blog](https://blog.tabulareditor.com/2024/08/14/tabular-editor-3-august-2024-release/)

In general, what you describe sounds like you are having authentication issues in the Power BI service especially regarding the gateway. But it is not something I have a whole lot of insight into setting up correctly.

**Reply by xMikeTR** [reply]
Wanted to thank you! I ended up connecting to Databricks, other sources and it worked.

I had to add it to my gateway in the end but it worked

**Reply by mlonsk** [reply]
Hi @xMikeTR 
You're welcome; I'm happy to hear you got it to work!
